Earthfile

```diff
@@ -1092,3 +1177,67 @@ bump-repositories:
     RUN yq eval ".repositories[0] |= . * { \"reference\": \"${REPO_AMD64}\" }" -i framework-profile.yaml
     RUN yq eval ".repositories[1] |= . * { \"reference\": \"${REPO_ARM64}\" }" -i framework-profile.yaml
     SAVE ARTIFACT framework-profile.yaml AS LOCAL framework-profile.yaml
+
+# Installs the needed bits for "standard" images (the provider ones)
+PROVIDER_INSTALL:
+    COMMAND
+
+    ARG PROVIDER_KAIROS_BRANCH
+
+    COPY +luet/luet /usr/bin/luet
+
+    IF [[ "$PROVIDER_KAIROS_BRANCH" = "" ]] # Install with luet (released versions of the binary)
+      # If base image does not bundle a luet config use one
+      # TODO: Remove this, use luet config from base images so they are in sync
+      IF [ ! -e "/etc/luet/luet.yaml" ]
+          COPY framework-profile.yaml /etc/luet/luet.yaml
+      END
+
+      # We don't specify a version. To bump, just change what the latest version
+      # in the repository is.
+      RUN luet install -y system/provider-kairos
+    ELSE # Install from a branch
+      COPY github.com/kairos-io/provider-kairos:$PROVIDER_KAIROS_BRANCH+build-kairos-agent-provider/agent-provider-kairos /system/providers/agent-provider-kairos
+      RUN ln -s /system/providers/agent-provider-kairos /usr/bin/kairos
+    END
+
+INSTALL_NOHANG:
+    COMMAND
+
+    ARG FLAVOR
+
+    # Install nohang
+    IF [ "$FLAVOR" = "opensuse-leap" ] || [ "$FLAVOR" = "opensuse-leap-arm-rpi" ]
+      RUN zypper ref && zypper in -y nohang
+    ELSE IF [ "$FLAVOR" = "alpine-ubuntu" ] || [ "$FLAVOR" = "alpine-opensuse-leap" ] || [ "$FLAVOR" = "alpine-arm-rpi" ]
+      RUN apk add grep
+    ELSE IF [ "$FLAVOR" = "opensuse-tumbleweed" ] || [ "$FLAVOR" = "opensuse-tumbleweed-arm-rpi" ]
+      RUN zypper ref && zypper in -y nohang
+    ELSE IF [ "$FLAVOR" = "ubuntu" ] || [ "$FLAVOR" = "ubuntu-20-lts" ] || [ "$FLAVOR" = "ubuntu-22-lts" ] || [ "$FLAVOR" = "debian" ]
+      RUN apt-get update && apt-get install -y nohang
+    END
+
+# Installs k3s (for "standard" images)
+INSTALL_K3S:
+    COMMAND
+
+    ARG K3S_VERSION
+    IF [[ "$K3S_VERSION" = "" ]]
+      RUN echo "$K3S_VERSION must be set" && exit 1
+    END
+
+    IF [ ! -e "/etc/luet/luet.yaml" ]
+        COPY framework-profile.yaml /etc/luet/luet.yaml
+    END
+
+    IF [[ "$K3S_VERSION" = "latest" ]] # Install latest using the upstream installer
+      ENV INSTALL_K3S_BIN_DIR="/usr/bin"
+      RUN curl -sfL https://get.k3s.io > installer.sh \
+          && INSTALL_K3S_SELINUX_WARN=true INSTALL_K3S_SKIP_START="true" INSTALL_K3S_SKIP_ENABLE="true" INSTALL_K3S_SKIP_SELINUX_RPM="true" bash installer.sh \
+          && INSTALL_K3S_SELINUX_WARN=true INSTALL_K3S_SKIP_START="true" INSTALL_K3S_SKIP_ENABLE="true" INSTALL_K3S_SKIP_SELINUX_RPM="true" bash installer.sh agent \
+          && rm -rf installer.sh
+    ELSE
+      ARG _LUET_K3S=$(k8s/k3s@${K3S_VERSION})
+    END
+
+    RUN luet install -y ${_LUET_K3S} utils/edgevpn utils/k9s utils/nerdctl container/kubectl utils/kube-vip && luet cleanup
```

Comment:
will this work if we go over the `[[ "$K3S_VERSION" = "latest" ]]` branch? It means that `${_LUET_K3S}` will be empty no? just woprried about luet misbehaving with empty spaces lol

Reply:
We tried `latest` and it worked. Actually we intentionally refactored this code to work for the `latest` case because initially the `latest` branch didn't install `k9s` et al.